test/SlowTests/Issues/RavenDB-22948.cs

```diff
@@ -0,0 +1,315 @@
+        }
+
+        [RavenFact(RavenTestCategory.Querying | RavenTestCategory.ClientApi)]
+        public void Test_TwoClients_DifferentIdHandlingWithLoad()
+        {
+            using (var defaultStore = new DocumentStore
+            {
+                Urls = new[] { Server.WebUrl },
+                Database = "Test"
+            }.Initialize())
+            {
+                var doc1Id = "users/1";
+                var doc2Id = "users/2";
+
+                using (var customStore = GetDocumentStore(new Options
+                {
+                    ModifyDatabaseName = x => "Test",
+                    ModifyDocumentStore = s =>
+                    {
+                        s.Conventions.FindIdentityPropertyNameFromCollectionName = (typeName) => "TestId";
+                        s.Conventions.FindIdentityProperty = prop => prop.Name == "TestId";
+                    }
+                }))
+                {
+                    //Id is identity property
+                    using (var session1 = defaultStore.OpenSession())
+                    {
+                        var defaultUser = new User
+                        {
+                            Id = doc1Id,
+                            TestId = "users/1-T",
+                            Name = "Default Store User"
+                        };
+                        session1.Store(defaultUser); //Document Id is users/1
+                        session1.SaveChanges();
+                    }
+
+                    //TestId is identity property
+                    using (var session2 = customStore.OpenSession())
+                    {
+                        var customUser = new User
+                        {
+                            Id = "users/2-T",
+                            TestId = doc2Id,
+                            Name = "Custom Store User"
+                        };
+                        session2.Store(customUser); //Document Id is users/2
+                        session2.SaveChanges();
+
+                        var user1 = session2.Load<User>(doc1Id);
+                        user1.Id = "users/3";
+                        session2.Store(user1); // changing Id property users/1 document to users/3
+                        session2.SaveChanges();
+                    }
+
+                    //Id is identity property
+                    using (var session1 = defaultStore.OpenSession())
+                    {
+                        var results = session1
+                            .Query<User>()
+                            .Select(x => new UserWithPet()
+                            {
+                                User = session1.Load<User>(x.Id),
+                                Pet = "My Pet"
+                            })
+                            .ToList();
+
+                        //*****************Not expected****************************
+                        Assert.Equal("users/2-T", results[0].User.Id); //need to be users/2
```

Comment:
But what if we want to load results.user.id? We won't get the actual document.
In this case, the Id is supposed to be the identity property, but the Id we get is not the real document Id.

I would expect that if I receive a document Id, I should be able to load the document using that Id.

Reply:
You need to use `RavenQuery.Id()` or something like that.
Because otherwise you'll step over the actual values.

Note the distinction between projected value and actual document here.